File: app/src/main/java/com/ciastek/todoapp/TaskState.java
package com.ciastek.todoapp;

public enum TaskState {
    NEW("New"),
    IN_PROGRESS("In progress"),
    DONE("Done");

    private String stateName;
    private int value;

    TaskState (String stateName) {
        this.stateName = stateName;
    }

    @Override
    public String toString () {
        return stateName;
    }

    public int getValue () {
        switch (stateName) {
            case "New":
                return 0;
            case "In progress":
                return 1;
            case "Done":
                return 2;
            default:
                return 0;
        }
    }
}

File: app/src/main/java/com/ciastek/todoapp/model/TaskState.kt
package com.ciastek.todoapp.model

import android.support.annotation.IntDef
import com.ciastek.todoapp.model.STATE.DONE
import com.ciastek.todoapp.model.STATE.IN_PROGRESS
import com.ciastek.todoapp.model.STATE.NEW

@Retention(AnnotationRetention.SOURCE)
@IntDef(NEW, IN_PROGRESS, DONE)
annotation class TaskState